src/main/scala/is/hail/expr/Parser.scala

```diff
@@ -710,12 +710,21 @@ object Parser extends JavaTokenParsers {
 
   def genomeReferenceDependentTypes: Parser[String] = "Variant" | "LocusInterval" | "Locus"
 
+  def intervalWithEndpoints[T](bounds: Parser[(T, T)]): Parser[Interval] = {
+    val start = ("[" ^^^ true) | ("(" ^^^ false)
+    val end = ("]" ^^^ true) | (")" ^^^ false)
+
+    start ~ bounds ~ end ^^ {case istart ~ int ~ iend => Interval(int._1, int._2, istart, iend)} |
+      bounds ^^ {int => Interval(int._1, int._2, true, false)}
```

Comment:
Although @cseed said to remove backwards compatibility for my other PR.

Reply:
out it goes, then.

Reply:
actually, I think it needs to exist to maintain compatibility with the .interval_list format. Where does that come from?

Reply:
ok I keep waffling back and forth on this, but I think I want to leave it like this for this PR. The biggest (only?) reason is that the open/closed bounds aren't really exposed on the python side, and Interval.parse() uses the parser to parse an interval. I'm happy to change the python-side interval stuff to require these bounds, also, but maybe we should discuss that further before making the change. In any case, I'd like to save that potential change (all the python stuff) for a separate PR, since I need this change for the partitioner stuff.

Reply:
ok. I just realized that BedAnnotator needs to be fixed. It assumes the endpoint is not included. Double check the python docs to ensure that assumption is being made.

Reply:
BedAnnotator looks like it should still behave the same. I haven't changed the default behavior of the interval constructor, although perhaps I should.